File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int n = strs.length;
        Arrays.sort(strs);
        
        StringBuilder ans = new StringBuilder();
        int i = 0;
        while(i<strs[0].length() && i<strs[n-1].length()){
            if(strs[0].charAt(i) == strs[n-1].charAt(i)){
                ans.append(strs[0].charAt(i));
            }else{
                break;
            }
            i++;
        }
        return ans.toString();
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans = "";
        
        for(int i = 0; i < strs[0].size(); i++){
            char ch = strs[0][i];
            for(int j = 0; j < strs.size(); j++){
                if(strs[j].size() <= i) return ans;
                
                if(strs[j][i] != ch) return ans;
            }
            ans.push_back(ch);
        }
        
        return ans;
    }
};